feat: 3.1 Spec - Add active problem status

angelathe opened the pull request:
# PULL REQUEST

## Summary

- Add problem status to Active Problems table
- Update snapshot test

## Related Issue

Fixes #603 

## Screenshot

**eCR has problem status**
<img width="758" alt="Screenshot 2025-07-01 at 12 31 06" src="https://github.com/user-attachments/assets/f6df786f-c4b2-42d7-834d-5375b8c8e55d" />

**eCR does not have problem status**
<img width="1512" alt="Screenshot 2025-07-01 at 12 32 15" src="https://github.com/user-attachments/assets/564022a5-b8bd-4dcd-9179-25ea1fa7cf61" />

## Acceptance Criteria

- [ ] Field should be added to the eCR Viewer
- [ ] Any relevant unit/snapshot tests should be added.

## Additional Information

Note: Made decision w/ Ashton to add status after the name/before Onset Date/Time (this is not the same as what's currently in the Figma)

Can view problem status with the 3.1 sample eCR

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

JNygaard-Skylight commented on containers/ecr-viewer/src/app/utils/evaluate/fhir-paths.ts:
```diff
@@ -107,6 +107,7 @@ export type PathTypes = {
   resolve: unknown;
   activeProblems: Condition;
   activeProblemsDisplay: string;
+  activeProblemsStatus: ValueX;
```
This should be a `string`, because you are selecting the `display`. `ValueX` is for `value[x]` fields that can have multiple types.

JNygaard-Skylight commented on containers/ecr-viewer/src/app/utils/evaluate/fhir-paths.ts:
```diff
@@ -463,6 +464,10 @@ const _fhirPathMappings: { [K in FhirPathKeys]: Omit<FhirPath<K>, "name"> } = {
     path: "Condition.code.coding.display.first()",
   },
+  activeProblemsStatus: {
+    type: "ValueX",
```
This should be `string`, to match the above comment.

JNygaard-Skylight commented on containers/ecr-viewer/src/app/utils/evaluate/fhir-paths.ts:
```diff
@@ -463,6 +464,10 @@ const _fhirPathMappings: { [K in FhirPathKeys]: Omit<FhirPath<K>, "name"> } = {
   },
+  activeProblemsStatus: {
+    type: "ValueX",
+    path: "Condition.clinicalStatus.coding[0].display",
```
If you are only expecting 1 coding , you can skip the `[0]`.

JNygaard-Skylight reviewed (changes requested):
The way FHIR path works is a little tricky, and I'm concerned this might not be doing what we want it to be doing. Can you either modify the snapshot test, create a new snapshot test, create a unit test, or whatever test you want to test when there are different values for active problem status?

angelathe commented:
Good thinking. I added another active problem to the Condition test bundle (with a different problem status), and added unit tests to ensure that 1) both problems show up, and 2) both unique problem statuses are displayed. LMK what you think!

JNygaard-Skylight commented:
> @JNygaard-Skylight Good thinking. I added another active problem to the Condition test bundle (with a different problem status), and added to a unit test to ensure that 1) both problems show up, and 2) both unique problem statuses are displayed. LMK what you think!

I think I would feel better if you were testing that the active problem values where in the correct rows, instead of just checking whether the value exists somewhere. That said I just checked an eCR with two different statuses, and it appears to be working, so this is good enough for me!
